test/distrib/bazel/test_single_bazel_version.sh

```diff
@@ -51,12 +51,14 @@ FAILED_TESTS=""
 export OVERRIDE_BAZEL_VERSION="$VERSION"
 # when running under bazel docker image, the workspace is read only.
 export OVERRIDE_BAZEL_WRAPPER_DOWNLOAD_DIR=/tmp
+
 bazel build -- //... "${EXCLUDED_TARGETS[@]}" || FAILED_TESTS="${FAILED_TESTS}Build "
 
 for TEST_DIRECTORY in "${TEST_DIRECTORIES[@]}"; do
   pushd "test/distrib/bazel/$TEST_DIRECTORY/"
 
-  bazel test --test_output=all //:all || FAILED_TESTS="${FAILED_TESTS}${TEST_DIRECTORY} Distribtest"
+  tools/bazel version | grep "$VERSION" || { echo "Detected bazel version did not match expected value of $VERSION" >/dev/stderr; exit 1; }
+  tools/bazel test --test_output=all //:all || FAILED_TESTS="${FAILED_TESTS}${TEST_DIRECTORY} Distribtest"
```

Comment:
we should really have a good explanation here, so that we are not making changes blindly.

Reply:
> it works when you do it from the repo root.
Not for me. I _thought_ it was supposed to work, but I can't get it to with Bazel 

```
rbellevi@rbellevi:~/dev/tmp/grpc$ cd /
rbellevi@rbellevi:/$ bazel version
WARNING: Invoking Bazel in batch mode since it is not invoked from within a workspace (below a directory having a WORKSPACE file).
Build label: 5.3.1
...
rbellevi@rbellevi:/$ cd -
/usr/local/google/home/rbellevi/dev/tmp/grpc
rbellevi@rbellevi:~/dev/tmp/grpc$ bazel version
Build label: 5.3.1
...
rbellevi@rbellevi:~/dev/tmp/grpc$ tools/bazel version
INFO: Running bazel wrapper (see //tools/bazel for details), bazel version 4.2.2 will be used instead of system-wide bazel installation.
Starting local Bazel server and connecting to it...
Build label: 4.2.2
...
```

I also tried changing my installed version to  both 4.X and 3.X. Neither work from the repo root. Did this ever actually work? 